ManualDeploymentExtended.md

```diff
@@ -0,0 +1,1014 @@
+
+##### 1. Install dependencies.
+```bash
+~$ sudo apt install nodejs build-essential npm
+```
+##### 2. Build RTL.
+```bash
+~$ cd ~/src
+~$ git clone https://github.com/ShahanaFarooqui/RTL.git
+~$ cd RTL
+~$ npm install
+```
+##### 3. Create a configuration file.
+A sample RTL.conf configuration file is available [here](https://github.com/ShahanaFarooqui/RTL/blob/master/sample-RTL.conf).
+```bash
+~$ vi RTL.conf
+```
+```text
+[SSO]
+rtlSSO=0
+rtlCookiePath=/var/lib/rtl
+logoutRedirectLink=/login
+
+[Authentication]
+macaroonPath=/var/lib/lnd/data/chain/bitcoin/mainnet
+lndConfigPath=/etc/lnd/lnd.conf
+nodeAuthType=CUSTOM
+rtlPass=password
+[Settings]
+flgSidenavOpened=true
+flgSidenavPinned=true
+menu=Vertical
+menuType=Regular
+theme=dark-blue
+satsToBTC=false
+bitcoindConfigPath=/etc/bitcoin/bitcoin.conf
+enableLogging=true
+port=3000
+lndServerUrl=https://localhost:8080/v1
+```
+Note that RTL has different behaviour and requirements compared to the other services documented in theses instructions, specifically:
+ 1. The configuration file needs to exist in RTL's data directory, 
+ 2. The RTL process writes to the configuration file. The main change it will do is convert any cleartext password set with `rtlPass` to a hashed version with `rtlPassHashed`.
+ 
+```bash
+~$ sudo mkdir -p /var/lib/rtl
+~$ sudo chown admin:admin -R /var/lib/rtl
+~$ sudo cp ~/RTL.conf /var/lib/rtl
+~$ sudo chmod 644 /var/lib/rtl/RTL.conf
+```
+##### 4. Create a systemd service.
+```bash
+~$ vi rtl.service
+```
+```text
+[Unit]
+Description=Ride The Lightning
+Requires=lnd.service
+After=lnd.service
+
+[Service]
+Environment="RTL_CONFIG_PATH=/var/lib/rtl/RTL.conf"
+WorkingDirectory=/var/lib/rtl
+ExecStart=/usr/bin/node /home/admin/src/RTL/rtl
+User=admin
+Group=admin
+Type=simple
+PIDFile=/run/rtl/rtl.pid
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+```
+```bash
+~$ sudo cp rtl.service /etc/systemd/system
+~$ sudo systemctl enable rtl
+~$ sudo systemctl start rtl
+```
+
+##### :thumbsup: Check
+
+Check the service:
+
+```bash
+~$ sudo journalctl -xe --unit rtl --follow
+...
+May 24 19:38:01 btc node[8072]: Console:
+May 24 19:38:01 btc node[8072]: Transactions: 10: 1558726681366: INFO: Transaction Received: {}
+```
+If it doesn't start correctly stop the service and run the application directly to get any error messages.
+```bash
+~$ sudo systemctl stop rtl
+~$ export RTL_CONFIG_PATH=/var/lib/rtl/RTL.conf; pushd ~/src/RTL; node rtl; popd;
+Server is up and running, please open the UI at http://localhost:3000
+```
+
+#####  :rotating_light: Update
+Updating could break things. Be careful on a live system.
+```bash
+~$ sudo systemctl stop rtl
+~$ cd ~; pushd ~/src/RTL; git pull; npm install; popd;
+~$ sudo systemctl start rtl
+```
+ 
+## :checkered_flag: The End
+
+That's all there is to it!
+
+If you were able to successfully complete the steps then you now have the equivalent of the multi-billion behemoth PayPal running on your own server.
```

Comment:
The last part is not necessary for technical doc, whoever is doing manual installation knows what he's installing